ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -213,13 +232,24 @@ public static BlockImportResult on_block(
       UnsignedLong finalized_slot = store.getFinalizedCheckpoint().getEpochStartSlot();
       // Update justified if new justified is later than store justified
       // or if store justified is not in chain with finalized checkpoint
+
+      boolean isFinalizedAncestorOfJustified;
+      try {
+        Bytes32 ancestorOfJustifiedAtFinalizedSlot =
+            get_ancestor(
+                forkChoiceStrategy, store.getJustifiedCheckpoint().getRoot(), finalized_slot);
+        isFinalizedAncestorOfJustified =
+            ancestorOfJustifiedAtFinalizedSlot.equals(store.getFinalizedCheckpoint().getRoot());
+      } catch (UnknownBlockRootException e) {
+        isFinalizedAncestorOfJustified = false;
+      }
+
       if (state
                   .getCurrent_justified_checkpoint()
                   .getEpoch()
                   .compareTo(store.getJustifiedCheckpoint().getEpoch())
               > 0
-          || !get_ancestor(store, store.getJustifiedCheckpoint().getRoot(), finalized_slot)
-              .equals(store.getFinalizedCheckpoint().getRoot())) {
+          || !isFinalizedAncestorOfJustified) {
```

Comment:
We should restore the order of these checks as finding the ancestor is a lot more expensive than checking the epochs.  So probably make `isFinalizedAncestorOfJustified` a method that's called here rather than a variable. Then we benefit from short circuiting.

Reply:
Makes sense. Done.